tests/python_package_test/test_dask.py

```diff
@@ -37,6 +47,134 @@ def listen_port():
 listen_port.port = 13000
 
 
+def _make_ranking(n_samples=100, n_features=20, n_informative=5, gmax=2,
+                  group=None, random_gs=False, avg_gs=10, random_state=0):
+    """Generate a learning-to-rank dataset - feature vectors grouped together with
+    integer-valued graded relevance scores. Replace this with a sklearn.datasets function
+    if ranking objective becomes supported in sklearn.datasets module.
+
+    Parameters
+    ----------
+    n_samples: int (default=100)
+        Total number of documents (records) in the dataset
+    n_features : int (default=20)
+        Total number of features in the dataset
+    n_informative : int (default=5)
+        Number of features that are "informative" for ranking, as they are bias + beta * unif(min=y, max=y+1),
+        where bias and beta are standard normal variates. If this is greater than n_features, the dataset will have
+        n_features features, all will be informative.
+    group : array-like, optional (default=None)
+        1-d array or list of group sizes. When `group` is specified, this overrides n_samples, random_gs, and
+        avg_gs by simply creating groups with sizes group[0], ..., group[-1].
+    gmax : int (default=2)
+        Maximum graded relevance value for creating relevance/target vector. If you set this to 2, for example, all
+        documents in a group will have relevance scores of either 0, 1, or 2.
+    random_gs : bool (default=False)
+        True will make group sizes ~ Poisson(avg_gs), False will make group sizes == avg_gs.
+    avg_gs : int (default=10)
+        Average number of documents (records) in each group
+
+    Returns
+    ----------
+    X : 2-d np.ndarray of shape = [n_samples (or np.sum(group), n_features]
+        Input feature matrix for ranking objective
+    y : 1-d np.array of shape = [n_samples (or np.sum(group))]
+        integer-graded relevance scores
+    group_ids: 1-d np.array of shape = [n_samples (or np.sum(group))]
+        vector of group ids, each value indicates to which group each record belongs
+    """
```

Comment:
```suggestion
    """Generate a learning-to-rank dataset - feature vectors grouped together with
    integer-valued graded relevance scores. Replace this with a sklearn.datasets function
    if ranking objective becomes supported in sklearn.datasets module.

    Parameters
    ----------
    n_samples : int, optional (default=100)
        Total number of documents (records) in the dataset.
    n_features : int, optional (default=20)
        Total number of features in the dataset.
    n_informative : int, optional (default=5)
        Number of features that are "informative" for ranking, as they are bias + beta * unif(min=y, max=y + 1),
        where bias and beta are standard normal variates. If this is greater than n_features, the dataset will have
        n_features features, all will be informative.
    group : array-like, optional (default=None)
        1-d array or list of group sizes. When `group` is specified, this overrides n_samples, random_gs, and
        avg_gs by simply creating groups with sizes group[0], ..., group[-1].
    gmax : int, optional (default=2)
        Maximum graded relevance value for creating relevance/target vector. If you set this to 2, for example, all
        documents in a group will have relevance scores of either 0, 1, or 2.
    random_gs : bool, optional (default=False)
        True will make group sizes ~ Poisson(avg_gs), False will make group sizes == avg_gs.
    avg_gs : int, optional (default=10)
        Average number of documents (records) in each group.

    Returns
    -------
    X : 2-d np.ndarray of shape = [n_samples (or np.sum(group)), n_features]
        Input feature matrix for ranking objective.
    y : 1-d np.array of shape = [n_samples (or np.sum(group))]
        Integer-graded relevance scores.
    group_ids: 1-d np.array of shape = [n_samples (or np.sum(group))]
        Array of group ids, each value indicates to which group each record belongs.
    """
```

Reply:
100%

Reply:
Hey heads up on this - I changed the `label` values from y[i] = unif(label[i], label[i] + 1) to make the test data simpler to get more reliable models on unit tests. So I'm changing ```Number of features that are "informative" for ranking, as they are bias + beta * unif(min=y, max=y + 1),```

To

```Number of features that are "informative" for ranking, as they are bias + beta * y```